### ISSUE TITLE
[🐛] Support accessing non-default databases 

### ISSUE THREAD

**Issue by ori9998** [invertase/react-native-firebase#7376]
## Issue
I've found that Flutter recently added support for multiple databases, and wondering whether react-native would also support this feature.

In iOS, Android, and Web code it is now possible to access non-default Firestore database instances by passing the database ID to the call that gets the Firestore instance. 

https://firebase.google.com/docs/reference/js/firestore_.md#getfirestore
According to the main docs,
export declare function getFirestore(app: FirebaseApp, databaseId: string): Firestore;
databaseId could be called and get the firestore instance.

Describe your issue here

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
# N/A
```

#### `firebase.json` for react-native-firebase v6:

```json
# N/A
```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ ] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
 OUTPUT GOES HERE
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `e.g. 5.4.3`
- **`Firebase` module(s) you're using that has the issue:**
  - `e.g. Instance ID`
- **Are you using `TypeScript`?**
  - `Y/N` & `VERSION`

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by ori9998** [comment]
inside index.js, 

/**
 * @param {FirebaseApp} app
 * @param {FirestoreSettings} settings
 * @param {string?} databaseId
 * @returns {Promise<Firestore>}
 */
export async function initializeFirestore(app, settings /* databaseId */) {
  // TODO(exaby73): implement 2nd database once it's supported
  const firestore = firebase.firestore(app);
  await firestore.settings(settings);
  return firestore;
}

databaseId is kinda implemented, but not sure what to do next..

**Comment by nyplex** [comment]
Any update on this?

**Comment by ori9998** [comment]
@exaby73 Just wondering, can I get any insight on this matter? Is this in high priority? I'm building a start up project based on this..

**Comment by mikehardy** [comment]
This is not something we have on our roadmap, but I would happily merge a PR that implements it.
I'm afraid I can't provide much direction for it though - as I've added multi-app support or region/database support to functions etc I've always just patterned off other similar modules which worked fine. As such looking at the functions package or the firestore package may be illuminating in how to plumb through an optional / non-default database name in order to enable the support here

**Comment by ribeiroantonio** [comment]
Hey guys, any update on this?

**Comment by MadhavKanna** [comment]
Hi would be great to have an update on this

**Comment by Ehesp** [comment]
Hi all - this landed a while ago, it looks like the issue got closed off via a bot so wasn't picked up in triage: https://github.com/invertase/react-native-firebase/pull/7949

TLDR is:

```js
firebase.firestore('second-db-id')...
```

**Comment by ktanoooo** [comment]
The option to specify a non-default database name has already been merged (https://github.com/invertase/react-native-firebase/pull/7949). 
It worked when I wrote it as follows.

```jsx
import { getApp } from '@react-native-firebase/app';
import { getFirestore } from '@react-native-firebase/firestore';

getFirestore(getApp(), 'yourDatabaseName')
  .collection('yourCollectionName')
  .doc('yourDocumentId')
  .onSnapshot((documentSnapshot) => {
    console.log('data: ', documentSnapshot.data());
  });
```
